### PR TITLE
Hotfix/6.1.0

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: src/main/java/com/iexec/sms/utils/EthereumCredentials.java [hunk 1]
```diff
@@ -42,33 +43,33 @@
...
-    private String address;
     private String privateKey;
-    private String publicKey;
-    private boolean isEncrypted; // private & public keys
+    private boolean isEncrypted;
 
-    public EthereumCredentials() throws Exception {
-        ECKeyPair ecKeyPair = Keys.createEcKeyPair();
-        this.address = Numeric.prependHexPrefix(Keys.getAddress(ecKeyPair));
-        setPlainKeys(toHex(ecKeyPair.getPrivateKey()),
```
Comment (jeremyjams) [inline src/main/java/com/iexec/sms/utils/EthereumCredentials.java:53]: Might produce non bytes32 hex string:

//0xab[...]a25b (size != 32)
//vs
//0x00ab[...]a25b (size == 32)
Dealing with full hex64 string (0x00ab[...]a25b) ensures stronger sanity checks on clients

File: src/main/java/com/iexec/sms/utils/EthereumCredentials.java [hunk 2]
```diff
@@ -42,33 +43,33 @@
     @GenericGenerator(name = "system-uuid", strategy = "uuid")
     private String id;
 
-    private String address;
     private String privateKey;
-    private String publicKey;
-    private boolean isEncrypted; // private & public keys
+    private boolean isEncrypted;
```
Comment (zguesmi) [inline src/main/java/com/iexec/sms/utils/EthereumCredentials.java:48]: Nice!

**Review by zguesmi — APPROVED** [review]

**Comment by zguesmi** [comment]
Thanks you for the clean-up.

File: src/main/java/com/iexec/sms/utils/EthereumCredentials.java [hunk 3]
```diff
@@ -42,33 +43,33 @@
...
+        this.setPlainTextPrivateKey(privateKey);
     }
 
-    public void setPlainKeys(String privateKey, String publicKey) {
-        this.setKeys(privateKey, publicKey, false);
+    public static EthereumCredentials generate() throws java.security.GeneralSecurityException {
+        ECKeyPair randomEcKeyPair = Keys.createEcKeyPair();
+        String privateKey =
+                Numeric.toHexStringWithPrefixZeroPadded(randomEcKeyPair.getPrivateKey(),
+                        Keys.PRIVATE_KEY_LENGTH_IN_HEX);//hex-string size of 32 bytes (64)
```
[the remark this answers is not in the document]
Comment (jeremyjams) [reply src/main/java/com/iexec/sms/utils/EthereumCredentials.java:65]: Added doc, thanks :+1:

File: src/main/java/com/iexec/sms/utils/EthereumCredentials.java [hunk 4]
```diff
@@ -45,21 +47,31 @@
 
     private String privateKey;
     private boolean isEncrypted;
+    /*
+     * Address is required since recovering from private key is not possible
+     * from an encrypted private key state.
```
Comment (zguesmi) [inline src/main/java/com/iexec/sms/utils/EthereumCredentials.java:52]: Is it to enable using this object without decrypting the private key?

Comment (jeremyjams) [reply src/main/java/com/iexec/sms/utils/EthereumCredentials.java:52]: Yes exactly

**Review by zguesmi — APPROVED** [review]

**Pull request merged**